c_wrapper/lua-mongoc-collection.c

```diff
@@ -650,53 +650,54 @@ lua_mongo_collection_create_index(lua_State *L)
     int keys_index = 3;
     int options_index = 4;
 
-    if (!(lua_istable(L, keys_index))) {
+    if(!(lua_istable(L, keys_index))){
         luaL_error(L, "keys parameter must be a table");
-    }
-
-    if (!(lua_isnil(L, keys_index))){
+    }else{
```

Comment:
If you do that, you should be able to remove the variable index_name, and just use `opt.name` everywhere